easybuild/framework/easyconfig/types.py

```diff
@@ -236,17 +247,47 @@ def to_dependency(dep):
         if not found_name_version:
             raise EasyBuildError("Can not parse dependency without name and version: %s", dep)
 
+    # also deal with dependencies in the "old" format
+    elif isinstance(dep, tuple):
+        depspec = dep
     else:
         raise EasyBuildError("Can not convert %s (type %s) to dependency dict", dep, type(dep))
 
     return depspec
 
 
-# this uses functions defined in this module, so it needs to be at the bottom of the module
+# these constants use functions defined in this module, so they needs to be at the bottom of the module
+
+# specific type: dict with only name/version as keys, and with string values
+# additional type requirements are specified as tuple of tuples rather than a dict, since this needs to be hashable
+NAME_VERSION_DICT = (dict, dict2tuple({
+    'only_keys': ('name', 'version'),
+    'value_types': (str,),
```

Comment:
with the proposed changes to `dict2tuple`, this can be:

```python
NAME_VERSION_DICT = (dict, dict2tuple({
    'only_keys': ['name', 'version'],
    'value_types': [str],
})
```